src/messages/MStatfs.h

```diff
@@ -35,17 +37,24 @@ class MStatfs : public PaxosServiceMessage {
 public:
   const char *get_type_name() const override { return "statfs"; }
   void print(ostream& out) const override {
-    out << "statfs(" << get_tid() << " v" << version << ")";
+    out << "statfs(" << get_tid() << " pool "
+        << (data_pool ? *data_pool : -1) << " v" << version << ")";
   }
 
   void encode_payload(uint64_t features) override {
     paxos_encode();
     ::encode(fsid, payload);
+    ::encode(data_pool, payload);
```

Comment:
It's encoded as a boost::optional<int>. There's a defined encoding for that:

https://github.com/ceph/ceph/blob/master/src/include/encoding.h#L289